tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/DeleteCombinationFeatureContext.php

```
@@ -38,28 +39,81 @@ class DeleteCombinationFeatureContext extends AbstractCombinationFeatureContext
      *
      * @param string $combinationReference
      */
-    public function deleteCombination(string $combinationReference): void
+    public function deleteCombinationInDefaultShop(string $combinationReference): void
     {
-        $this->getCommandBus()->handle(new DeleteCombinationCommand(
-            (int) $this->getSharedStorage()->get($combinationReference)
-        ));
+        $this->deleteSingleCombination($combinationReference, $this->getDefaultShopId());
     }
 
     /**
      * @When I delete following combinations of product :productReference:
      *
-     * @param string $productRefeence
+     * @param string $productReference
+     * @param TableNode $tableNode
+     */
+    public function bulkDeleteCombinationsInDefaultShop(string $productReference, TableNode $tableNode): void
+    {
+        $this->bulkDeleteCombinations($productReference, $tableNode);
+    }
+
+    /**
+     * @When I delete following combinations of product :productReference from shop :shopReference:
+     *
+     * @param string $productReference
+     * @param string $shopReference
+     * @param TableNode $tableNode
+     */
+    public function bulkDeleteCombinationsInShop(string $productReference, string $shopReference, TableNode $tableNode): void
+    {
+        $this->bulkDeleteCombinations($productReference, $tableNode, $shopReference);
+    }
+
+    /**
+     * @When I delete combination :combinationReference from shop :shopReference
+     *
+     * @param string $combinationReference
+     * @param string $shopReference
+     */
+    public function deleteCombinationInShop(string $combinationReference, string $shopReference): void
+    {
+        $this->deleteSingleCombination($combinationReference, $this->getSharedStorage()->get($shopReference));
+    }
+
+    /**
+     * @param string $combinationReference
+     * @param int|null $shopId
+     */
+    private function deleteSingleCombination(string $combinationReference, ?int $shopId): void
```

```suggestion
    private function deleteSingleCombination(string $combinationReference, int $shopId): void
```

---

I don't see why the `shopId` should be nullable, if it's empty it means we probably did something wrong at some point In which case I prefer the code to fail early